Geometry_Engine/Query/GeometryHash.cs

```diff
@@ -0,0 +1,417 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Security.Cryptography.Xml;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Returns the geometrical identity of any geometry primitive, useful for comparisons and diffing. " + 
+        "Any transformations of underlying geometry performed in methods as part of the calculation of the Geometry Hash " +
+        "are implemented to be translational only to ensure validity of any geometrical tolerance checking between two hashes downstream.")]
+        public static double[] IGeometryHash(this IBHoMObject bhomObj)
```

Comment:
Shouldn't this be accepting `IGeometry` rather than `IBHoMObject`? Think we should handle the IBHoMObject outside the Geometry_Engine (imo, there should not be any methods accepting any type of BHoMObejct in the Geometry_Engine)

Reply:
The reason I went for this solution:
1. I thought we'd like to expose `GeometryHash()` on top of every object, like we do for `Hash()` so my first choice was actually `IObject`
1. However, `IGeometry()` only accepts `IBHoMObject` and `CustomObject`. I agree with you, I don't like this either: this conflicts with 
   > imo, there should not be any methods accepting any type of BHoMObejct in the Geometry_Engine) 
1. I also am not sure we want to expose `GeometryHash()` for `IGeometry` directly: I think `GeometryHash()` makes more sense when paired with an object that is not geometry, because if you want to compare two geometries you can just do that. But don't feel strongly about this one, happy to revise.

Therefore, I went for `IBHoMObject`. I'd be more happy to change this to `IObject` honestly, so we replicate what `Hash()` targets.